extensions/compose/src/compose-extension.ts

```diff
@@ -0,0 +1,207 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+import * as path from 'node:path';
+import { existsSync, promises } from 'node:fs';
+import * as extensionApi from '@podman-desktop/api';
+import type { Detect } from './detect';
+import type { ComposeGitHubReleases } from './compose-github-releases';
+import type { OS } from './os';
+import { platform, arch } from 'node:os';
+import type { PodmanComposeGenerator } from './podman-compose-generator';
+
+export class ComposeExtension {
+  public static readonly COMPOSE_INSTALL_COMMAND = 'compose.install';
+  public static readonly COMPOSE_CHECKS_COMMAND = 'compose.checks';
+
+  public static readonly ICON_CHECK = 'fa fa-check';
+  public static readonly ICON_DOWNLOAD = 'fa fa-download';
+  public static readonly ICON_WARNING = 'fa fa-exclamation-triangle';
+
+  private statusBarItem: extensionApi.StatusBarItem | undefined;
+
+  protected currentInformation: string;
+
+  constructor(
+    private readonly extensionContext: extensionApi.ExtensionContext,
+    private readonly detect: Detect,
+    private readonly composeGitHubReleases: ComposeGitHubReleases,
+    private readonly os: OS,
+    private podmanComposeGenerator: PodmanComposeGenerator,
+  ) {}
+
+  async runChecks(firstCheck: boolean): Promise<void> {
+    this.currentInformation = undefined;
+
+    // reset status bar information
+    const statusBarChangesToApply = {
+      iconClass: '',
+      tooltip: '',
+      command: ComposeExtension.COMPOSE_CHECKS_COMMAND,
+    };
+
+    // detect if podman-compose is installed
+    const podmanComposeInstalled = await this.detect.checkForPythonPodmanCompose();
+
+    // in that case, report an issue in status bar
+    if (podmanComposeInstalled) {
+      // create a status bar item
+      statusBarChangesToApply.iconClass = ComposeExtension.ICON_WARNING;
+      this.currentInformation =
+        'This extension does not work with Python Podman Compose. It will collide with the CLI named podman-desktop. You need to uninstall Python Podman Compose before using Docker Compose v2.';
```

Comment:
```suggestion
        'This extension does not work with the Python implementation of the Compose spec (podman-compose). It conflicts with Podman Desktop. Uninstall podman-compose (ex. pip uninstall python-compose) before continuing.';
```

Maybe something like this would be better?

Reply:
I've removed the stuff for podman-compose v1. we don't need it anymore